src/System.Runtime.Serialization.Xml/tests/XmlCanonicalizationTest.cs

```diff
@@ -0,0 +1,453 @@
+                    continue;
+                }
+
+                string myDefaultNamespace = "http://mynamespace";
+                string myNamespace1 = "http://mynamespace1";
+                string myNamespace2 = "http://mynamespace2";
+                string myNamespace3 = "http://mynamespace3";
+                string myNamespace4 = "http://mynamespace4";
+                writer.WriteStartElement("Root");
+                writer.WriteXmlnsAttribute("p1", myNamespace1);
+                writer.WriteAttributeString("p1", "a", null, "b");
+                writer.WriteStartElement("", "Element1", myDefaultNamespace);
+                writer.WriteAttributeString("p3", "c", myNamespace3, "d");
+                writer.WriteStartElement("Element2");
+
+                MemoryStream canonicalStream = new MemoryStream();
+
+                writer.StartCanonicalization(canonicalStream, false, new string[] { "p3", "p2", "p1", "" });
+                writer.WriteStartElement("pre", "Element3", myNamespace2);
+                writer.WriteAttributeString("pre2", "attrName", myNamespace4, "attrValue");
+                writer.WriteStartElement("Element4", "");
+                writer.WriteStartAttribute("attr1");
+                writer.WriteQualifiedName("foo", myNamespace1);
+                writer.WriteEndAttribute();
+
+                writer.WriteStartAttribute("attr2");
+                writer.WriteQualifiedName("bar", myNamespace3);
+                writer.WriteEndAttribute();
+
+                writer.WriteString("Hello world");
+
+                writer.WriteEndElement(); // Element4
+                writer.WriteEndElement(); // pre:Element3
+
+                writer.EndCanonicalization();
+                writer.WriteEndElement(); // Element2
+                writer.WriteEndElement(); // Element1
+                writer.WriteEndElement(); // Root
+                writer.Flush();
+
+                byte[] canonicalDoc = canonicalStream.ToArray();
+                byte[] fullDoc = ms.ToArray();
+
+                writer.Close(); // Finished creating the document
+
+                XmlDsigExcC14NTransform transform = new XmlDsigExcC14NTransform();
+                transform.InclusiveNamespacesPrefixList = "p3 p2 p1 #default";
+                transform.LoadInput(new MemoryStream(canonicalDoc));
+                Stream transformedOutput = transform.GetOutput(typeof(Stream)) as Stream;
+                byte[] outputFromSecurity = StreamToByteArray(transformedOutput);
+                //Finished creating the doc from the security class
+
+                Helper.DumpToFile("fullDoc.xml", fullDoc);
+                Helper.DumpToFile("canonicalDoc.xml", canonicalDoc);
+                Helper.DumpToFile("docFromSecurity.xml", outputFromSecurity);
+
+                try
+                {
+                    Helper.CompareArrays(outputFromSecurity, 0, canonicalDoc, 0, canonicalDoc.Length);
```

Comment:
Can we do `Assert.True(Enumerable.SequenceEqual(outputFromSecurity, canonicalDoc))` and remove `Helper.CompareArrays`? We probably won't need the `try...catch...`